hub/api/dataset.py

```diff
@@ -30,7 +30,7 @@ def _generate_storage_tensors(self):
         for t in self._flat_tensors:
             t: FlatTensor = t
             yield t.path, StorageTensor(
-                f"{self.url}/{t.path}",
+                f"{self.url}{t.path}",
```

Comment:
This is a delicate situation, on one side os.path.join does not work properly on Windows when we deal with cloud url, on the other side to avoid double /-ing we need some kind of library or write it ourselves. You assume self.url ends with / and I don't. Both cases should be handled.

Reply:
Agree, I noticed that your paths had double `//` that's why did this, but the custom join is preferable.